packages/dds/tree/src/simple-tree/treeApi.ts

```diff
@@ -142,11 +176,82 @@ export const treeNodeApi: TreeNodeApi = {
 
 /**
  * A collection of events that can be raised by a {@link TreeNode}.
+ *
+ * @privateRemarks
+ * TODO: add a way to subscribe to a specific field (for afterShallowChange and afterDeepChange).
+ * Probably have object node and map node specific APIs for this.
+ *
+ * TODO: ensure that subscription API for fields aligns with API for subscribing to the root.
+ *
+ * TODO: add more wider area (avoid needing tons of afterShallowChange registration) events for use-cases other than afterDeepChange.
+ * Some ideas:
+ *
+ * - afterDeepChange, but with some subtrees/fields/paths excluded
+ * - helper to batch several afterShallowChange calls to a afterDeepChange scope
+ * - parent change (ex: registration on the parent field for a specific index: maybe allow it for a range. Ex: node event takes optional field and optional index range?)
+ * - new content inserted into subtree. Either provide event for this and/or enough info to afterDeepChange to find and search the new sub-trees.
+ * Add separate (non event related) API to efficiently scan tree for given set of types (using low level cursor and schema based filtering)
+ * to allow efficiently searching for new content (and initial content) of a given type.
+ *
  * @public
  */
-export interface TreeNodeEvents {
+export interface TreeChangeEvents {
 	/**
-	 * Raised on a node right after a change is applied to one of its fields or the fields of a descendant node.
+	 * Emitted by a node when:
+	 *
+	 * - The content of one or more of the node's properties changes (i.e., the property now contains a new node, or
```

Comment:
The "one or more" part helps convey that the event may be batched, but it is currently only called out for object nodes, so it looks as though array and map node updates are not batched.

You could fix this by updating the bullet points for map and array nodes, but I think it would be better to have the information that this event is batched be called out once at the top level, in its own sentence.

Reply:
How does this read?

```
* Emitted by a node when a batch of changes is applied to it, where a change is:
*
* - For an object node, when the value of one of its properties changes (i.e., the property's value is set
* to something else, including `undefined`).
*
* - For an array node, when an element is added, removed, or moved.
*
* - For a map node, when an entry is added, updated, or removed.
*
```

Reply:
I'm happy with that.